tests/acceptance/test_nidcpower_measurement.py

```diff
@@ -0,0 +1,106 @@
+    MeasurementServiceStub,
+)
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.pin_map_context_pb2 import (
+    PinMapContext,
+)
+from ni_measurementlink_service.measurement.service import MeasurementService
+from tests.assets.nidcpower_measurement_pb2 import NIDCPowerConfigurations, NIDCPowerOutputs
+from tests.utilities import nidcpower_measurement
+from tests.utilities.pin_map_client import PinMapClient
+
+
+def test___single_session___measure___single_session_created(
+    pin_map_client: PinMapClient,
+    pin_map_directory: pathlib.Path,
+    stub_v2: MeasurementServiceStub,
+) -> None:
+    pin_map_name = "1Smu1ChannelGroup1Pin1Site.pinmap"
+    pin_names = ["Pin1"]
+    pin_map_id = pin_map_client.update_pin_map(pin_map_directory / pin_map_name)
+    pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0])
+    configurations = NIDCPowerConfigurations(pin_names=pin_names, multi_session=False)
+
+    outputs = _measure(stub_v2, pin_map_context, configurations)
+
+    assert _get_output(outputs) == [
+        _MeasurementOutput("DCPower1/0", "DCPower1/0", "DCPower1/0", "DCPower1/0")
+    ]
+
+
+def test___multiple_sessions___measure___multiple_sessions_created(
+    pin_map_client: PinMapClient,
+    pin_map_directory: pathlib.Path,
+    stub_v2: MeasurementServiceStub,
+) -> None:
+    pin_map_name = "1Smu2ChannelGroup2Pin1Site.pinmap"
+    pin_names = ["Pin1", "Pin2"]
+    pin_map_id = pin_map_client.update_pin_map(pin_map_directory / pin_map_name)
+    pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0])
+    configurations = NIDCPowerConfigurations(pin_names=pin_names, multi_session=True)
+
+    outputs = _measure(stub_v2, pin_map_context, configurations)
+
+    assert _get_output(outputs) == [
+        _MeasurementOutput("DCPower1/0", "DCPower1/0", "DCPower1/0", "DCPower1/0"),
+        _MeasurementOutput("DCPower1/2", "DCPower1/2", "DCPower1/2", "DCPower1/2"),
+    ]
+
+
+def _measure(
+    stub_v2: MeasurementServiceStub,
+    pin_map_context: PinMapContext,
+    configurations: NIDCPowerConfigurations,
+) -> NIDCPowerOutputs:
+    request = MeasureRequest(pin_map_context=pin_map_context)
+    request.configuration_parameters.Pack(configurations)
+    response_iterator = stub_v2.Measure(request)
+    responses = list(response_iterator)
+    assert len(responses) == 1
+    outputs = NIDCPowerOutputs.FromString(responses[0].outputs.value)
+    return outputs
+
+
+@pytest.fixture(scope="module")
+def measurement_service() -> Generator[MeasurementService, None, None]:
+    """Test fixture that creates and hosts a measurement service."""
+    with nidcpower_measurement.measurement_service.host_service() as service:
+        yield service
+
+
+@pytest.fixture
+def pin_map_directory(test_assets_directory: pathlib.Path) -> pathlib.Path:
+    """Test fixture that returns the pin map directory."""
+    return test_assets_directory / "acceptance" / "session_management"
```

Comment:
If we are going to share assets between tests, we should be deliberate about it.

Move these new tests to `tests/acceptance/session_management/` and extract this test fixture into `tests/acceptance/session_management/conftest.py` so that all of the tests in this directory can share it. Likewise for the integration tests.

FYI, https://dev.azure.com/ni/DevCentral/_wiki/wikis/AppCentral.wiki/936/Using-Test-Assets recommends organizing test assets by test so that changes to one asset don't affect other tests unexpectedly, but I don't mind sharing pin maps as long as their naming scheme is clear enough to prevent unexpected changes.

Reply:
I've moved the fixtures in "test_nidcpower_measurement.py" and "test_nidcpower_reservation.py" to `tests/acceptance/conftest.py` and `tests/integration/conftest.py` respectively because I noticed that all the existing pin map assets are within a "session_management" directory already (`/assets/acceptance/session_management/` and `/assets/integration/session_management/`).

If I must have the `conftest.py` in the session_management folder within the `tests/acceptance/` and `tests/integration/` directory, then I think we should bring out the existing pin map files from the `assets/acceptance/session_management/` and `assets/integration/session_management/` directory to avoid any confusion in the future.

Please let me know your thoughts, @bkeryan.

Reply:
You can move test_reservation.py and test_session_management.py into the session_management directory if that helps.

Our regression test conventions specify that unit/integration tests should be organized following the structure of the code under test. That structure recently changed. Originally, the code under test had a session_management.py module, but I split it up into a session_management package with private submodules, including session_management/_reservation.py.

Reply:
I've restructured the test files as per your suggestion. 👍🏼